### ISSUE TITLE
Improve support for Dark Theme

### ISSUE THREAD

**Issue by eric-milles** [checkstyle/eclipse-cs#98]
With default color settings, Checkstyle markers look pretty bad under the Dark Theme (Window > Preferences > General > Appearance > Theme).  I'm not sure if I can actually modify these colors or if they are chosen within the Checkstyle plug-ins code.

![checkstyle](https://user-images.githubusercontent.com/18193802/44231162-c93ec080-a162-11e8-8ee0-d06c90bd0b31.png)

**Comment by vorburger** [comment]
I don't see Checkstyle in menu Window > Preferences : General > Appearances > Colours and Fonts, so it's probably hard-coded in the code... do you fancy contributing to the project :smile: by making this configurable?

**Comment by lkoe** [comment]
It's configurable under Preferences: General > Editors > Text Editors > Annotations
There are 3 Checkstyle related annotations definitons (error, warning, info) which can be adapted.

As for supporting the dark theme out of the box, any hints how this could be done?

**Comment by eric-milles** [comment]
Yes, to add support for the switchover, you can add an extension like this: https://github.com/groovy/groovy-eclipse/blob/master/ide/org.codehaus.groovy.eclipse.ui/plugin.xml#L1211


The idea is to add pseudo-selectors in the CSS that match your preference names.

[https://avatars3.githubusercontent.com/u/64846?s=400&v=4]<https://github.com/groovy/groovy-eclipse/blob/master/ide/org.codehaus.groovy.eclipse.ui/plugin.xml#L1211>

groovy/groovy-eclipse<https://github.com/groovy/groovy-eclipse/blob/master/ide/org.codehaus.groovy.eclipse.ui/plugin.xml#L1211>
github.com
groovy-eclipse - Groovy-Eclipse




There is a longer treatment of it here: https://www.eclipse.org/community/eclipse_newsletter/2018/june/darktheme.php (see Guide for plug-in developers to contribute to the default dark theme)

[https://www.eclipse.org/eclipse.org-common/themes/solstice/public/images/logo/eclipse-foundation-200x200.png]<https://www.eclipse.org/community/eclipse_newsletter/2018/june/darktheme.php>

Dark Theme in Eclipse Photon | The Eclipse Foundation<https://www.eclipse.org/community/eclipse_newsletter/2018/june/darktheme.php>
www.eclipse.org
With Eclipse Photon, we are shipping the best support for the dark theme so far!





________________________________

**Comment by cad2298** [comment]
What I have done is to set the marker as "vertical bar" instead of "highlighted" on Window > Preferences > General > Editors > Text Editors and now it does not look such annoying as before.
